packages/cactus-plugin-ledger-connector-corda/src/main/typescript/api-client/corda-api-client.ts

```diff
@@ -0,0 +1,152 @@
+import { Observable, ReplaySubject } from "rxjs";
+import { finalize, share } from "rxjs/operators";
+import {
+  Logger,
+  LogLevelDesc,
+  LoggerProvider,
+  Checks,
+} from "@hyperledger/cactus-common";
+import { ISocketApiClient } from "@hyperledger/cactus-core-api";
+import {
+  DefaultApi,
+  GetMonitorTransactionsV1ResponseTx,
+} from "../generated/openapi/typescript-axios";
+import { Configuration } from "../generated/openapi/typescript-axios/configuration";
+
+const DEFAULT_POOL_RATE_MS = 5000;
+
+type CordaBlock = GetMonitorTransactionsV1ResponseTx;
+
+export type watchBlocksV1Options = {
+  readonly stateFullClassName: string;
+  readonly pollRate?: number;
+};
+
+export class CordaApiClientOptions extends Configuration {
+  readonly logLevel?: LogLevelDesc;
+}
+
+export class CordaApiClient
+  extends DefaultApi
+  implements ISocketApiClient<CordaBlock> {
+  public static readonly CLASS_NAME = "CordaApiClient";
+
+  private readonly log: Logger;
+
+  public get className(): string {
+    return CordaApiClient.CLASS_NAME;
+  }
+
+  constructor(public readonly options: CordaApiClientOptions) {
+    super(options);
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.log.debug(`Created ${this.className} OK.`);
+    this.log.debug(`basePath=${this.options.basePath}`);
+  }
+
+  private sendStartMonitorRequest(
+    subject: ReplaySubject<GetMonitorTransactionsV1ResponseTx>,
+    stateName: string,
+  ) {
+    const reportError = (err: any) => {
+      this.log.warn("Error in startMonitorV1:", err);
+      subject.error(`startMonitorV1 for '${stateName}' transactions failed`);
+    };
+
+    this.startMonitorV1({
+      stateFullClassName: stateName,
+    })
+      .then((startMonRes) => {
+        if (startMonRes.status != 200 || !startMonRes.data.success) {
+          reportError(
+            `Wrong response: status ${startMonRes.status}, success ${startMonRes.data.success}`,
+          );
+        } else {
+          this.log.info(`Monitoring for ${stateName} transactions started.`);
+        }
+      })
+      .catch((err) => {
+        reportError(err);
+      });
+  }
+
+  private async poolTransactionsLogin(
```

Comment:
@outSH Is it pooling or polling? Or is it both, e.g. polling the pool?

Reply:
@petermetz Same as above